spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/convert/TwoStepsConversions.java

```diff
@@ -205,11 +249,23 @@ public Value convertOnWrite(Object proppertyVal, DatastorePersistentProperty per
 		return writeConverter.apply(val);
 	}
 
-	private EntityValue convertOnWriteSingleEmbedded(Object val, String kindName) {
+	private EntityValue convertOnWriteSingleEmbedded(Object val, String kindName,
+			boolean isEmbeddedComponents) {
 		IncompleteKey key = this.objectToKeyFactory.getIncompleteKey(kindName);
 
 		FullEntity.Builder<IncompleteKey> builder = FullEntity.newBuilder(key);
-		this.datastoreEntityConverter.write(val, builder);
+		// Write an embedded Map
```

Comment:
move this out to its own function so we have symmetry of 3 and 3 converter functions for read and write